src/display/annotation_layer.js

```diff
@@ -38,341 +38,515 @@ var CustomStyle = displayDOMUtils.CustomStyle;
+        iconName.toLowerCase() + '.svg';
+      image.alt = '[{{type}} Annotation]';
+      image.dataset.l10nId = 'text_annotation_type';
+      image.dataset.l10nArgs = JSON.stringify({type: iconName});
+
+      var contentWrapper = document.createElement('div');
+      contentWrapper.className = 'annotTextContentWrapper';
+      contentWrapper.style.left = Math.floor(rect[2] - rect[0] + 5) + 'px';
+      contentWrapper.style.top = '-10px';
+
+      var content = this.content = document.createElement('div');
+      content.className = 'annotTextContent';
+      content.setAttribute('hidden', true);
 
-    return container;
-  }
+      var i, ii;
+      if (this.data.hasBgColor && this.data.color) {
+        var color = this.data.color;
 
-  function getHtmlElement(data, page, viewport, linkService) {
-    switch (data.annotationType) {
-      case AnnotationType.WIDGET:
-        return getHtmlElementForTextWidgetAnnotation(data, page, viewport);
-      case AnnotationType.TEXT:
-        return getHtmlElementForTextAnnotation(data, page, viewport);
-      case AnnotationType.LINK:
-        return getHtmlElementForLinkAnnotation(data, page, viewport,
-                                               linkService);
-      default:
-        throw new Error('Unsupported annotationType: ' + data.annotationType);
-    }
-  }
+        // Enlighten the color (70%).
+        var BACKGROUND_ENLIGHT = 0.7;
+        var r = BACKGROUND_ENLIGHT * (255 - color[0]) + color[0];
+        var g = BACKGROUND_ENLIGHT * (255 - color[1]) + color[1];
+        var b = BACKGROUND_ENLIGHT * (255 - color[2]) + color[2];
+        content.style.backgroundColor = Util.makeCssRgb(r | 0, g | 0, b | 0);
+      }
+
+      var title = document.createElement('h1');
+      var text = document.createElement('p');
+      title.textContent = this.data.title;
 
-  function render(viewport, div, annotations, page, linkService) {
-    for (var i = 0, ii = annotations.length; i < ii; i++) {
-      var data = annotations[i];
-      if (!data || !data.hasHtml) {
-        continue;
+      if (!this.data.content && !this.data.title) {
+        content.setAttribute('hidden', true);
+      } else {
+        var e = document.createElement('span');
+        var lines = this.data.content.split(/(?:\r\n?|\n)/);
+        for (i = 0, ii = lines.length; i < ii; ++i) {
+          var line = lines[i];
+          e.appendChild(document.createTextNode(line));
+          if (i < (ii - 1)) {
+            e.appendChild(document.createElement('br'));
+          }
+        }
+        text.appendChild(e);
+
+        var self = this;
+        image.addEventListener('click', function image_clickHandler() {
+          self._toggle();
+        }, false);
+        image.addEventListener('mouseover', function image_mouseOverHandler() {
+          self._show();
+        }, false);
+        image.addEventListener('mouseout', function image_mouseOutHandler() {
+          self._hide();
+        }, false);
```

Comment:
Done in the new commit. Note that I needed to add the `isBool` check and import because with this oneliner the event object is now passed to the listener, where we first expected only to get a boolean value.